cmd/fluxctl/await.go

```diff
@@ -69,24 +70,24 @@ func awaitJob(client api.ClientService, jobID job.ID) (history.CommitEventMetada
 }
 
 // await polls for a commit to have been applied, with exponential backoff.
-func awaitSync(client api.ClientService, revision string) error {
+func awaitSync(ctx context.Context, client api.Client, revision string) error {
 	return backoff(1*time.Second, 2, 10, 1*time.Minute, func() (bool, error) {
-		refs, err := client.SyncStatus(noInstanceID, revision)
+		refs, err := client.SyncStatus(ctx, revision)
 		return err == nil && len(refs) == 0, err
 	})
 }
 
 // backoff polls for f() to have been completed, with exponential backoff.
 func backoff(initialDelay, factor, maxFactor, timeout time.Duration, f func() (bool, error)) error {
 	maxDelay := initialDelay * maxFactor
-	finish := time.Now().UTC().Add(timeout)
-	for delay := initialDelay; time.Now().UTC().Before(finish); delay = min(delay*factor, maxDelay) {
```

Comment:
_en passant_ -- I don't know why these included a call to UTC() -- paranoia perhaps. But: unnecessary and thereby mildly misleading.